subprojects/core/src/main/java/org/gradle/process/internal/DefaultExecActionFactory.java

```diff
@@ -123,9 +123,7 @@ public JavaForkOptionsInternal newDecoratedJavaForkOptions() {
 
     @Override
     public JavaForkOptionsInternal newJavaForkOptions() {
-        final DefaultJavaForkOptions forkOptions = new DefaultJavaForkOptions(fileResolver, fileCollectionFactory, new DefaultJavaDebugOptions());
-        forkOptions.setExecutable(Jvm.current().getJavaExecutable());
-        return forkOptions;
+        return new DefaultJavaForkOptions(fileResolver, fileCollectionFactory, new DefaultJavaDebugOptions());
```

Comment:
Reverting this change but not the one to `newDecoratedJavaForkOptions`. Is there a specific reason?